hack/make/containerutility

```diff
@@ -8,6 +8,9 @@ CONTAINER_UTILITY_COMMIT=e004a1415a433447369e315b9d7df357102be0d2 # v0.9.0
 	cd "$GOPATH/src/github.com/docker/windows-container-utility"
 	git checkout -q "$CONTAINER_UTILITY_COMMIT"
 
+	# TODO fix container-utility Makefile to take both gcc-mingw-w64 6.3 (stretch) and 8.3 (buster) into account
+	sed -i 's|-I/usr/lib/gcc/x86_64-w64-mingw32/6.3-win32/include|-I/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include -I/usr/lib/gcc/x86_64-w64-mingw32/6.3-win32/include|g' "$GOPATH/src/github.com/docker/windows-container-utility/Makefile"
```

Comment:
I was wondering why these paths have to be specified manually, because these paths should be known already;

```
x86_64-w64-mingw32-g++ -print-search-dirs
install: /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/
programs: =/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/bin/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/bin/
libraries: =/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/lib/x86_64-w64-mingw32/8.3-win32/:/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/lib/
```


Looking further, I noticed that the makefile sets `-nostdinc`, which disables MinGW searching in the standard paths;

Without specifying any path manually, and with `` set;

```bash
x86_64-w64-mingw32-g++ -nostdinc -v -fno-exceptions -std=c++11 -static -DUNICODE -municode -O3 -s -o containerutility.exe argumentstream.cpp commands.cpp containerutility.cpp identity.cpp version.cpp

GNU C++11 (GCC) version 8.3-win32 20190406 (x86_64-w64-mingw32)
	compiled by GNU C version 8.3.0, GMP version 6.1.2, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.20-GMP

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
#include "..." search starts here:
#include <...> search starts here:
End of search list.
GNU C++11 (GCC) version 8.3-win32 20190406 (x86_64-w64-mingw32)
	compiled by GNU C version 8.3.0, GMP version 6.1.2, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.20-GMP

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
Compiler executable checksum: ae7a3589826a0028176d9a8a3489295b
In file included from identity.cpp:1:
containerutility.h:3:21: error: no include path in which to search for windows.h
 #include <windows.h>
                     ^
```

Removing that option makes compilation succeed without any manual path specified;

```bash
x86_64-w64-mingw32-g++ -fno-exceptions -std=c++11 -static -DUNICODE -municode -O3 -s -o containerutility.exe argumentstream.cpp commands.cpp containerutility.cpp identity.cpp version.cpp
echo $?
0

Reply:
So, the only path not included in the defaults is `/usr/x86_64-w64-mingw32/include`. That one looks to be containing symlinks to `/usr/share/mingw-w64/include/`

```
ls -la /usr/x86_64-w64-mingw32/include | more
total 136
drwxr-xr-x 10 root root 69632 Jan  7 11:40 .
drwxr-xr-x  5 root root  4096 Jan  7 11:40 ..
drwxr-xr-x  2 root root  4096 Jan  7 11:40 GL
lrwxrwxrwx  1 root root    42 Nov 11  2018 _bsd_types.h -> ../../share/mingw-w64/include/_bsd_types.h
lrwxrwxrwx  1 root root    39 Nov 11  2018 _cygwin.h -> ../../share/mingw-w64/include/_cygwin.h
lrwxrwxrwx  1 root root    38 Nov 11  2018 _dbdao.h -> ../../share/mingw-w64/include/_dbdao.h
lrwxrwxrwx  1 root root    38 Nov 11  2018 _mingw.h -> ../../share/mingw-w64/include/_mingw.h
lrwxrwxrwx  1 root root    47 Nov 11  2018 _mingw_dxhelper.h -> ../../share/mingw-w64/include/_mingw_dxhelper.h
```

Reply:
Removing the `-nostdinc`, but _adding_ that path will make it compile, but looks like the path will be ignored because it overlaps with a system directory;

```bash
x86_64-w64-mingw32-g++ -v -fno-exceptions -I/usr/x86_64-w64-mingw32/include -std=c++11 -static -DUNICODE -municode -O3 -s -o containerutility.exe argumentstream.cpp commands.cpp containerutility.cpp identity.cpp version.cpp

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
ignoring nonexistent directory "/usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/sys-include"
ignoring duplicate directory "/usr/x86_64-w64-mingw32/include"
  as it is a non-system directory that duplicates a system directory
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include/c++
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include/c++/x86_64-w64-mingw32
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include/c++/backward
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/include-fixed
 /usr/lib/gcc/x86_64-w64-mingw32/8.3-win32/../../../../x86_64-w64-mingw32/include
End of search list.
```